### DISCUSSION TITLE
cordova-plugin-file: resolveLocalFileSystemURI() usage for 'content:/' URLs

### DISCUSSION THREAD

**Discussion by iamrakesh** [apache/cordova#369] [Q&A] [answered]
Hi,

Is FileUtils#resolveLocalFileSystemURI() implementation in cordova-plugin-file expected to work for url strings with 'content://' protocol ?

What I observe is that #resolveNativeUri() call at https://github.com/apache/cordova-plugin-file/blob/master/src/android/FileUtils.java#L743 first converts given 'content://' url into 'https://localhost/__cdvfile_content__/.....' format and later at ContentFileSystem#exists() method call at https://github.com/apache/cordova-plugin-file/blob/master/src/android/FileUtils.java#L752, internally tries to create 'content://' url back (see https://github.com/apache/cordova-plugin-file/blob/master/src/android/ContentFilesystem.java#L123), here implementation results invalid url in the form of 'content://e_content__/....' and subsequently fails with an IOException

Is it allowed to use #resolveLocalFileSystemURI() for 'content://' urls ?
Is there any other way ?

Thank you

**Comment by breautek** [comment, marked as answer]
Based on what you've found, I think this is probably a bug and probably should be reported as an issue inside cordova-plugin-file.

**Reply by iamrakesh** [reply]
Thank you for confirming.
I now see an issue has been reported already and a PR open for this issue. Below are the details
Issue: https://github.com/apache/cordova-plugin-file/issues/525
Pull request: https://github.com/apache/cordova-plugin-file/pull/534

I see some tests are failing and the logs are not available anymore to check the details.
